src/main/java/org/gaul/httpbin/HttpBin.java

```diff
@@ -17,36 +17,72 @@
 
 package org.gaul.httpbin;
 
-import static java.util.Objects.requireNonNull;
-
-import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
 
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.HttpConfiguration;
 import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.SecureRequestCustomizer;
 import org.eclipse.jetty.server.Server;
 import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.SslConnectionFactory;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
 
 /**
  * Reimplementation of HttpBin https://httpbin.org/ suitable for offline unit
  * tests.
  */
 public final class HttpBin {
     private final Server server;
+    private final int mHTTPPort;
+    private final int mHTTPsPort;
 
-    public HttpBin(URI endpoint) throws Exception {
-        this(endpoint, new HttpBinHandler());
+    public HttpBin(String ip, int httpPort, int httpsPort, String keystore) throws Exception {
+        this(ip, httpPort, httpsPort, keystore, new HttpBinHandler());
     }
 
-    public HttpBin(URI endpoint, HttpBinHandler handler) throws Exception {
-        requireNonNull(endpoint);
+    public HttpBin(String ip, int httpPort, int httpsPort, String keystore, HttpBinHandler handler) throws Exception {
 
         server = new Server();
         HttpConnectionFactory httpConnectionFactory =
                 new HttpConnectionFactory();
-        ServerConnector connector = new ServerConnector(server,
+
+        mHTTPPort = httpPort;
+        mHTTPsPort = httpsPort;
+        List<Connector> connectors = new ArrayList<Connector>();
+        if (httpPort != 0) {
+            ServerConnector connector = new ServerConnector(server,
                 httpConnectionFactory);
-        connector.setHost(endpoint.getHost());
-        connector.setPort(endpoint.getPort());
-        server.addConnector(connector);
+            connector.setHost(ip);
+            connector.setPort(httpPort);
+            connectors.add(connector);
+        }
+
+        if (httpsPort != 0) {
+            HttpConfiguration https = new HttpConfiguration();
+            SecureRequestCustomizer src = new SecureRequestCustomizer();
+            src.setSniHostCheck(false);
+            https.addCustomizer(src);
+            SslContextFactory.Server sslContextFactory = new SslContextFactory.Server();
+            sslContextFactory.setKeyStorePath(keystore);
+            sslContextFactory.setKeyStorePassword("123456");
+            sslContextFactory.setKeyManagerPassword("123456");
```

Comment:
Is this the best way to configure the keystore?  Shouldn't it come from some external source like S3Proxy does?